tiledb/common/status.h

```diff
@@ -89,6 +89,14 @@ namespace common {
     }                            \
   } while (false)
 
+#define RETURN_NOT_OK_TUPLE_2(s)               \
```

Comment:
Hmm I don't really like the name here. Do we need this? If a function errors out, I don't this returning two `std::nullopt` would be useful. @eric-hughes-tiledb perhaps can chime in.

Reply:
Made this return an optional pair.